CentralComputing/Sensor_Test.cpp

```diff
@@ -0,0 +1,388 @@
+#include "Sensor.h"
+#include <iostream>
+#include <unistd.h>
+#include "Pod.h"
+#include "StateMachineCompact/stdafx.h"
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <vector>
+#include <stdlib.h>
+
+#include <time.h>
+
+using namespace std;
+
+// declaring function, implented later
+void printHelp();
+void printHelpInstr();
+
+/**
+	brings out the best in all of us
+*/
+void memes() {
+	srand(time(NULL));
+	int r = rand() % 4 +1;
+	cout << "MEME LORD IS HERE FOR YOU:" << endl;
+	if(r == 1) {
+		cout << "http://media.treehugger.com/assets/images/2013/08/elon-musk-hyperloop-good-guy-elon-musk-meme-chris-tackett.png.662x0_q70_crop-scale.png" << endl;
+	} else if (r == 2) {
+		cout << "http://s2.quickmeme.com/img/e6/e6b0f2395a1b7168c30c19881f51e8d20e83d08bec4d0b2a67e69a0cd18b43b1.jpg" << endl;
+	} else if (r == 3) {	
+		cout << "http://www.mememaker.net/static/images/memes/4284241.jpg" << endl;
+	} else if (r == 4) {
+		cout << "https://cdn.meme.am/cache/instances/folder695/63313695.jpg" << endl;
+	}
+	cout << endl;
+}
+
+/**
+	checks if the current state of pod matches witht the check state command
+	word: check state command to be executed
+	pod: reference of pod to be checked
+
+	return: whether the command returned true or false
+*/
+bool checkState(string word, Pod& pod) {
+	size_t start = word.find("[");
+	size_t end = word.find("]");
+	string arg = word.substr(start+1,end-1);
+	int state = pod.getCurrentState();
+
+	// pos handles if (current_state == some_state) should return true or false
+	bool pos = true;
+	if(arg.front() == '!') {
+		// negates the positive claim
+		pos = false;
+		// removes ! from argument
+		arg = arg.substr(1);
+	}
+	
+	// switch case for all the different states
+	// uses
+	switch(state) {
```

Comment:
Ok. I was thinking it would be useful for the tests you described where we see if calling a command like coast while in a state like functional_tests wouldn't then move to coast.